include/field3d.hxx

```diff
@@ -226,30 +228,42 @@ class Field3D : public Field, public FieldData {
   
   /// Check if this field has yup and ydown fields
   bool hasYupYdown() const {
-    return (yup_field != nullptr) && (ydown_field != nullptr);
+    return !yup_fields.empty() and !ydown_fields.empty();
```

Comment:
If we added a `bool hasValidYUpDown`, could be `return hasValidYUpDown`, since if `mergeYupYdown()` has been called, then calling `yup()`, etc., should be allowed so this should return true even if `yup_fields` is empty.

I'd been deleting yup/ydown fields on #1176, but now I've changed that to calling `setHasValidYUpDown(false)`. 4a7cbcfc9b80db86b6454552cb003569d7ffa0d4 is meant to have all the places that yup/ydown should be marked as invalid.

Reply:
:+1: am I right in thinking that the `std::vector` will not deallocate when we call `clear`, so we're not actually losing anything by deleting them?

Reply:
`std::vector::clear` erases its contents, so the values will be lost but it does retain its capacity though, so next time we create the parallel slices we won't need to allocate any new memory, if that's what you meant.